app/scripts/directives/registrationTypeSelect.js

```diff
@@ -6,8 +6,20 @@ angular.module('confRegistrationWebApp')
       templateUrl: template,
       restrict: 'E',
       controller: function ($scope, $rootScope, $location, $routeParams, RegistrationCache, uuid, modalMessage) {
+
         $scope.visibleRegistrantTypes = angular.copy($scope.conference.registrantTypes);
 
+        const findCurrentGroupRegistrantType = function() {
```

Comment:
Could you do something like this?
```js
const findCurrentGroupRegistrantType = (registrants, registrantTypes) => {
  const registrantTypeIds = registrants.map(
    ({ registrantTypeId }) => registrantTypeId
  );
  return registrantTypes.find(
    ({ allowGroupRegistrations, registrantTypeIds }) =>
      allowGroupRegistrations && registrantTypeIds.includes(type.id)
  );
};
```
I think what you've got is a little hard to read with 2 types of loops, having them nested, and the extra conditions.

Reply:
I agree, thanks for the tip!